src/kv/flatbufferwrapper.h

```diff
@@ -6,41 +6,85 @@
 
 namespace kv
 {
-  class FlatbufferSerialiser
+  namespace frame
   {
-  private:
-    flatbuffers::FlatBufferBuilder builder;
-    flatbuffers::Offset<Frame> frame;
-
-  public:
-    FlatbufferSerialiser(
-      const std::vector<uint8_t>& replicated,
-      const std::vector<uint8_t>& derived)
+    // These static functions provide access to the Frame internals without the
+    // need to create a deserialiser
+    static const Frame* root(const uint8_t* data)
     {
-      auto fb_replicated = builder.CreateVector(replicated);
-      auto fb_derived = builder.CreateVector(derived);
+      return GetFrame(data);
+    }
 
-      frame = CreateFrame(builder, fb_replicated, fb_derived);
-      builder.Finish(frame);
+    static const std::pair<const uint8_t*, size_t> replicated(
+      const uint8_t* data)
+    {
+      auto frame = GetFrame(data);
+      return {frame->replicated()->Data(), frame->replicated()->size()};
     }
 
-    flatbuffers::DetachedBuffer get_flatbuffer()
+    static const std::pair<const uint8_t*, size_t> derived(const uint8_t* data)
```

Comment:
Can we return `CBuffer` or similar here (with names for fields) rather than a pair?

Reply:
sure works nicely with CBuffer